relay-metrics/src/aggregation.rs

```diff
@@ -2720,97 +2766,91 @@ mod tests {
     #[test]
     fn test_flush_bucket() {
         relay_test::setup();
+        let system = actix::System::current();
+        let (tx, rx) = oneshot::channel();
+        let rt = tokio::runtime::Runtime::new().unwrap();
+
         let receiver = TestReceiver::default();
-        relay_test::block_fn(|| {
-            let config = AggregatorConfig {
-                bucket_interval: 1,
-                initial_delay: 0,
-                debounce_delay: 0,
-                ..Default::default()
-            };
-            let recipient = receiver.clone().start().recipient();
-            let project_key = ProjectKey::parse("a94ae32be2584e0bbd7a4cbb95971fee").unwrap();
-            let aggregator = Aggregator::new(config, recipient).start();
+        let config = AggregatorConfig {
```

Comment:
From an offline conversation: If you move `config`, `recipient`, `project_key` into `rt.spawn(async move {})`, your diff will look much cleaner. 

Basically, all that has to change in this test is that you replace `relay_test::block_fn` with the new runtime, along with a channel.

I would propose to add a simple utility for this, similar to:

```rust
fn block_with_actix(f: impl Future<Output = ()>) {
    let system = actix::System::current();
    let (tx, rx) = tokio01::sync::oneshot::channel();
    
    let rt = tokio::runtime::Builder::new_multi_thread()
        .worker_threads(1)
        .enable_all()
        .build()
        .unwrap();

    rt.spawn(async move {
        actix::System::set_current(system);
        compat::init();
        f.await;
        tx.send(()).ok();
    });

    // Note that this also resolves with error if `f` panics.
    relay_test::block_fn(|| rx).ok();
}
```